tests/test_docs.py

```diff
@@ -0,0 +1,34 @@
+from pathlib import Path
+import subprocess
+import re
+import logging
+import pytest
+import os
+
+if not os.getenv("CI") or os.getenv("SKIP_DOCS_TEST", False):
+    pytest.skip("Skipping docs test", allow_module_level=True)
+
+
+logging.getLogger("acquire").setLevel(logging.CRITICAL)
+
+DOCS_REPO = "https://github.com/acquire-project/acquire-docs"
+CODE_BLOCK = re.compile(r"```python\n(.*?)```", re.DOTALL)
+SKIP = {
+    "setup.md",  # has invalid syntax
+    "trigger.md",  # has some non-existant paths
+}
+
+# NOTE: this clones the repo on import... not the best practice, could be improved
+if not (DOCS_PATH := Path("acquire-docs", "docs")).exists():
+    subprocess.check_call(["git", "clone", DOCS_REPO])
```

Comment:
Could make this a function, e.g.,

```python
def docs_repo():
    skip = {
        "setup.md",  # has invalid syntax
        "trigger.md",  # has some non-existant paths
    }
    docs_repo = "https://github.com/acquire-project/acquire-docs"

    if not (docs_path := Path("acquire-docs", "docs")).exists():
        subprocess.check_call(["git", "clone", docs_repo])

    tuts = [docs_path / "get_started.md"]
    tuts.extend([fn for fn in docs_path.glob("tutorials/*.md") if fn.name not in skip])
    tuts.sort()

    return tuts
```
and call it in the decorator for `test_tutorials()`. I tried it out and it seems to work. Here's what my whole file looks like:

```python
from pathlib import Path
import subprocess
import re
import logging
import pytest

logging.getLogger("acquire").setLevel(logging.CRITICAL)

code_block = re.compile(r"```python\n(.*?)```", re.DOTALL)


def docs_repo():
    print("calling docs_repo()")
    skip = {
        "setup.md",  # has invalid syntax
        "trigger.md",  # has some non-existant paths
    }
    docs_repo = "https://github.com/acquire-project/acquire-docs"

    if not (docs_path := Path("acquire-docs", "docs")).exists():
        subprocess.check_call(["git", "clone", docs_repo])

    tuts = [docs_path / "get_started.md"]
    tuts.extend([fn for fn in docs_path.glob("tutorials/*.md") if fn.name not in skip])
    tuts.sort()

    return tuts


@pytest.mark.parametrize("tutorial", docs_repo(), ids=lambda x: x.name)
def test_tutorials(tutorial: Path):
    for block in code_block.finditer(tutorial.read_text()):
        exec(block.group(1))
```

Reply:
It would be great if we could make it a fixture that would clean up the docs repo after yielding, but I don't know if that's possible, to call a fixture from the decorator like that.

Reply:
Failing that, an entry in the `.gitignore` for `tests/acquire-docs`. Would hate to accidentally commit it with a `git add .`.

Reply:
No, on further reflection we probably want tests in both repos.

Reply:
ok, sounds good.  I'm away teaching a course for another week, but will have another look at the patterns used here and update stuff in a bit

Reply:
@tlambert03 Were you ever able to take another look at this?

Reply:
sorry, not yet.  will have a look  today.

a quick note on your suggestion above though: creating a function that gets called inside the call to `parametrize (`@pytest.mark.parametrize("tutorial", docs_repo(), ids=lambda x: x.name)`   is pretty much the same as defining the list returned by `docs_repo` at module level.  It doesn't "delay" that until it's needed by the fixture.  The primary challenge here is to both parametrize it on each tutorial *and* delay cloning and cleanup the docs repo as a regular fixture (because pytest needs params to be declared at test collection time, not at run time).  the alternative is to just run a for loop inside of a single test... which would certainly make for a cleaner test directory pattern, but would prevent you from testing each tut individually